WordPress/src/main/java/org/wordpress/android/ui/media/MediaSourceWPImages.java

```diff
@@ -174,54 +160,76 @@ private void addWordPressImagesFromCursor(Cursor cursor) {
             } while (cursor.moveToNext());
 
             removeDeletedEntries();
-        } else {
-            mLoading = false;
-            notifyLoadingStatus();
+        } else if (mListener != null) {
+            mListener.onMediaLoaded(true);
         }
     }
 
     private void removeDeletedEntries() {
-        AsyncTask<List<MediaItem>, Void, Void> backgroundCheck = new AsyncTask<List<MediaItem>, Void, Void>() {
-            @Override
-            protected Void doInBackground(List<MediaItem>[] params) {
-                for (MediaItem mediaItem : params[0]) {
-                    try {
-                        URL mediaUrl = new URL(mediaItem.getSource().toString());
-                        HttpURLConnection connection = (HttpURLConnection) mediaUrl.openConnection();
-                        connection.setRequestMethod("GET");
-                        connection.connect();
-                        int responseCode = connection.getResponseCode();
-
-                        if (responseCode == 200) {
-                            mVerifiedItems.add(mediaItem);
+        List<MediaItem> existingItems = new ArrayList<>(mMediaItems);
+
+        for (MediaItem mediaItem : existingItems) {
+            final boolean callLoaded = existingItems.indexOf(mediaItem) == existingItems.size() - 1;
+
+            AsyncTask<MediaItem, Void, MediaItem> backgroundCheck = new AsyncTask<MediaItem, Void, MediaItem>() {
+                @Override
+                protected MediaItem doInBackground(MediaItem[] params) {
+                    MediaItem mediaItem = params[0];
+                        try {
+                            URL mediaUrl = new URL(mediaItem.getSource().toString());
+                            HttpURLConnection connection = (HttpURLConnection) mediaUrl.openConnection();
+                            connection.setRequestMethod("GET");
+                            connection.connect();
+                            int responseCode = connection.getResponseCode();
+
+                            if (responseCode == 200) {
+                                mVerifiedItems.add(mediaItem);
```

Comment:
This may be overkill, but I try to avoid modifying an adapter's underlying data in the background like this. Instead, I have `doInBackground` work with a temporary list that's copied to the underlying data in `onPostExecute` ([example](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2414-empty-media-view-squashed/WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderTagAdapter.java#L229)).